pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -625,6 +626,12 @@ func (m *migrationMonitor) startMonitor() {
 			if strings.Contains(m.migrationFailedWithError.Error(), "canceled by client") {
 				abortStatus = v1.MigrationAbortSucceeded
 			}
+			// Improve the error message when the volume migration fails because the destination size is smaller then the source volume
+			if len(vmi.Status.MigratedVolumes) > 0 && errors.Is(m.migrationFailedWithError, errors.New("has to be smaller or equal to the  actual size of the containing file")) {
```

Comment:
I'm sorry, when I asked about using a typed error, I was wondering if we were getting a specific error back. If that is not the case, then I think your previous solution works better. See: https://go.dev/play/p/Bude3pxDUCv

Reply:
Reverted the change